luarules/gadgets/game_no_rush_mode.lua

```diff
@@ -72,6 +72,17 @@ if gadgetHandler:IsSyncedCode() then
 
 		if frame < norushtimer and (not TeamIDsToExclude[unitTeam]) then
 			local _, _, _, _, _, allyTeamID = Spring.GetTeamInfo(unitTeam)
+			if cmdID == CMD.INSERT then
+			  cmdID = cmdParams[2]
+			  local s = #cmdParams == 7 and 4 or 3
+			  local e = #cmdParams
+			  for i = 1, s do
+				cmdParams[i] = cmdParams[i + 3]
+			  end
+			  for i = s + 1, e do
+				cmdParams[i] = nil
+			  end
```

Comment:
Is there a less cryptic way to accomplish this?

Reply:
What is `s`? What is `e`? What on earth is this line doing `#cmdParams == 7 and 4 or 3`? How does someone reading this code know what it does?